### PR TITLE
Upgrade open-interpreter to 0.1.7

### PR DESCRIPTION
**Pull request by orange-crow**
open-interpreter 0.1.7 uses generator methods in respond method to replace recursion to avoid infinite recursion bug.

### CODE REVIEW

File: metagpt/tools/code_interpreter.py [hunk 1]
```diff
@@ -41,13 +41,14 @@ def __init__(self, auto_run: bool = True) -> None:
...
         interpreter.api_key = CONFIG.openai_api_key
-        interpreter.api_base = CONFIG.openai_api_base
+        # interpreter.api_base = CONFIG.openai_api_base
         self.interpreter = interpreter
 
     def chat(self, query: str, reset: bool = True):
         if reset:
             self.interpreter.reset()
-        return self.interpreter.chat(query, return_messages=True)
+        # return self.interpreter.chat(query, return_messages=True)
```
Comment (stellaHSR) [inline metagpt/tools/code_interpreter.py:50]: maybe just remove this

File: metagpt/tools/code_interpreter.py [hunk 2]
```diff
@@ -61,11 +62,19 @@ def extract_function(query_respond: List, function_name: str, *, language: str =
             assert language == 'python', f"Expect python language for default function_format, but got {language}."
             function_format = """def {function_name}():\n{code}"""
         # Extract the code module in the open-interpreter respond message.
-        code = [item['function_call']['parsed_arguments']['code'] for item in query_respond
-                if "function_call" in item
-                and "parsed_arguments" in item["function_call"]
-                and 'language' in item["function_call"]['parsed_arguments']
-                and item["function_call"]['parsed_arguments']['language'] == language]
+        if "function_call" in query_respond[1]:
```
Comment (stellaHSR) [inline metagpt/tools/code_interpreter.py:65]: maybe you can provide an data example of the query_respond here in the comment

**Comment by stellaHSR** [comment]
LGTM

**Review by LeonZh0u — APPROVED** [review]
LGTM

File: requirements.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@ pydantic==1.10.8
 #pymilvus==2.2.8
 pytest==7.2.2
 python_docx==0.8.11
-PyYAML==6.0
```
Comment (LeonZh0u) [inline requirements.txt:26]: can you keep the pinged version? add the version of "pip freeze |grep <pkg>" back into requirement.txt

File: requirements.txt [hunk 4]
```diff
@@ -39,13 +39,12 @@ typing_extensions==4.5.0
 libcst==1.0.1
 qdrant-client==1.4.0
 pytest-mock==3.11.1
-open-interpreter==0.1.4; python_version>"3.9"
+open-interpreter==0.1.7; python_version>"3.9"
 ta==0.10.2
-semantic-kernel==0.3.10.dev0
+semantic-kernel
```
Comment (stellaHSR) [inline requirements.txt:44]: 需要保持sk这个版本

**Pull request merged**